managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3177,8 +3177,7 @@ protected void asyncCreateLedger(BookKeeper bookKeeper, ManagedLedgerConfig conf
                 digestType, config.getPassword(), cb, ledgerCreated, finalMetadata);
         scheduledExecutor.schedule(() -> {
             if (!ledgerCreated.get()) {
-                ledgerCreated.set(true);
-                cb.createComplete(BKException.Code.TimeoutException, null, null);
+                cb.createComplete(BKException.Code.TimeoutException, null, ledgerCreated);
```

Comment:
Wouldn't there be a race condition here (also in existing code) if the creation completes at the same time that the timeout is triggered?

Reply:
The callback is synchronized at ManagedLedgerImpl. So there is no race condition here.

Reply:
It's not always the case. eg: In `createNewMetadataLedger()` when we create a new cursor ledger, we pass a lambda and there's no mutex. The logic in `checkAndCompleteLedgerOpTask()` seems race prone.

I think we should probably use compareAndSet() to manipulate that `ledgerCreated` flag, which also might be better represented by an enum with multiple states. 

Finally (and this is unrelated to the bug here), we should try to cancel the timeout task, once the create operation succeeds or fails before it times out. Otherwise we can get into a state with many timer tasks pending.

Reply:
I see. I missed the createNewMetadataLedger operation. I changed it to use CAS.

For canceling the timeout tasks, I will have a follow-up issue to do that.